src/lock_file/update.rs

```diff
@@ -668,7 +669,9 @@ impl<'p> UpdateContextBuilder<'p> {
         let project = self.project;
         let package_cache = match self.package_cache {
```

Comment:
I think we can simplify this part by using combinators

`let package_cache = self.package_cache.or_else(|| pixi_config::get_cache_dir()?.join(consts::CONDA_PACKAGE_CACHE_DIR))`